Neo4j.Driver/Neo4j.Driver/Internal/ConnectionValidator.cs

```diff
@@ -48,6 +49,8 @@ internal class ConnectionValidator : IConnectionValidator
         _livenessTimeout = livenessCheckTimeout.HasValue
             ? (long)livenessCheckTimeout.Value.TotalMilliseconds
             : long.MaxValue;
+
+        _idleTimerActive = _connIdleTimeout < long.MaxValue || _livenessTimeout < long.MaxValue;
```

Comment:
You're changes make sense, but I'm wondering about the initial logic that was implemented that you've built on top of. Without more context it looks like it is considering a timeout active if it is set to zero?